lib/auth/clt.go

```diff
@@ -70,6 +70,21 @@ const (
 // Dialer defines dialer function
 type Dialer func(network, addr string) (net.Conn, error)
 
+// IClientServices defines wrapper methods for roundtrip package.
+// Methods that are not wrappers are dependent on the roundtrip.Client instance
+// held by ClientService struct (which implements this interface).
+//
+// Interface was made for testing backward compatability logic between GRPC and REST.
+type IClientServices interface {
```

Comment:
Don't prefix type names with the kind of type (like `I` here). Also, this type doesn't need to be exported.
How about `type rpcClient`? Then `ClientServices` below can be renamed to `restRPCClient`.

Reply:
revised, but had a question, is rest a type of rpc? I searched the internet and they seem to distinguish between the two

Reply:
technically, REST is more of an API design concept.
but in practice it almost always means: "RPC over HTTP with URLs and HTTP verbs identifying the kind of operation to be performed".
so I tend to think of REST as a kind of RPC transport layer.
an alternative name here would be `httpRPCClient`